### PR TITLE
chore: [CDS-71756]: remove usage of CDS_AWS_NATIVE_LAMBDA FF

### PR DESCRIPTION
**Pull request by chetanvekariya-harness**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Review by pankajsingh-harness — APPROVED** [review]

**Comment by chetanvekariya-harness** [comment]
retrigger cypress-rest

**Comment by chetanvekariya-harness** [comment]
retrigger jest

**Comment by chetanvekariya-harness** [comment]
retrigger coverage

**Review by rameshx — APPROVED** [review]

**Review by PratyushGar — APPROVED** [review]

**Comment by chetanvekariya-harness** [comment]
retrigger jest

**Review by ishantmehtaharness — APPROVED** [review]
looks good

File: src/modules/75-cd/components/PipelineStudio/DeployServiceSpecifications/SelectDeploymentType/SelectDeploymentType.tsx [hunk 1]
```diff
@@ -269,19 +269,17 @@ export default function SelectDeploymentType({
   const { getString } = useStrings()
   const formikRef = React.useRef<FormikProps<unknown> | null>(null)
   const { subscribeForm, unSubscribeForm } = React.useContext(StageErrorContext)
-  const { NG_SVC_ENV_REDESIGN, CDS_TAS_NG, CDS_GOOGLE_CLOUD_FUNCTION, CDS_AWS_NATIVE_LAMBDA, CDP_AWS_SAM } =
-    useFeatureFlags()
+  const { NG_SVC_ENV_REDESIGN, CDS_TAS_NG, CDS_GOOGLE_CLOUD_FUNCTION, CDP_AWS_SAM } = useFeatureFlags()
 
   // Supported in NG (Next Gen - The one for which you are coding right now)
   const ngSupportedDeploymentTypes = React.useMemo(() => {
```
[the remark this answers is not in the document]
Comment (chetanvekariya-harness) [reply src/modules/75-cd/components/PipelineStudio/DeployServiceSpecifications/SelectDeploymentType/SelectDeploymentType.tsx:275]: Existing code, Ishant. I can revisit it later.

**Comment by chetanvekariya-harness** [comment]
retrigger jest

File: src/modules/75-cd/components/PipelineStudio/DeployServiceSpecifications/SelectDeploymentType/SelectDeploymentType.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (ishantmehtaharness) [reply src/modules/75-cd/components/PipelineStudio/DeployServiceSpecifications/SelectDeploymentType/SelectDeploymentType.tsx:275]: Sure chetan

**Pull request merged**